### PR TITLE
feat: create apis for advance analytics enrollments

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Description:** Add API endpoints for advance analytics enrollments data.

**JIRA:**
* https://2u-internal.atlassian.net/browse/ENT-9230
* https://2u-internal.atlassian.net/browse/ENT-9233

---

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/openedx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_data/admin_analytics/constants.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""Advanced Analytics Constants"""
+
+# granularity choices
+GRANULARITY_DAILY = 'Daily'
```
Comment (saleem-latif) [inline enterprise_data/admin_analytics/constants.py]: nit: I think enums here would be more readable.

```
class Granularity(enum.Enum):
    """
    Enum for the different granularity options.
    """
    daily = 'Daily'
    weekly = 'Weekly'
    monthly = 'Monthly'
    quarterly = 'Quarterly'
```

File: enterprise_data/admin_analytics/utils.py [hunk 2]
```diff
@@ -19,6 +27,54 @@ class ChartType(Enum):
     TOP_SKILLS_COMPLETION = 'top_skills_completion'
 
 
+def granularity_aggregation(level, group, date, data_frame, aggregation_type="count"):
+    """Aggregate data based on granularity"""
```
Comment (saleem-latif) [inline enterprise_data/admin_analytics/utils.py:24]: Please add more details in the doc string along with `Arguments` and `Returns` sections.

**Review by saleem-latif — APPROVED** [review]

File: enterprise_data/admin_analytics/utils.py [hunk 3]
```diff
@@ -19,6 +20,54 @@ class ChartType(Enum):
...
+
+    df.columns = group + [agg_column_name]
+    return df
+
+
+def calculation_aggregation(calc, data_frame, aggregation_type="count"):
+    """Aggregate data based on calculation"""
+    df = data_frame
+
+    window_mapping = {
```
Comment (irfanuddinahmad) [inline enterprise_data/admin_analytics/utils.py:51]: window?

File: enterprise_data/api/v1/views/analytics_enrollments.py [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+            ]
+        ]
+        enrollments["enterprise_enrollment_date"] = enrollments["enterprise_enrollment_date"].dt.date
+        enrollments = enrollments.sort_values(by="enterprise_enrollment_date", ascending=False).reset_index(drop=True)
+
+        if csv_type == ENROLLMENT_CSV.INDIVIDUAL_ENROLLMENTS.value:
+            return StreamingHttpResponse(
+                IndividualEnrollmentsCSVRenderer().render(self._stream_serialized_data(enrollments)),
+                content_type="text/csv",
+                headers={"Content-Disposition": 'attachment; filename="individual_enrollments.csv"'},
```
Comment (irfanuddinahmad) [inline enterprise_data/api/v1/views/analytics_enrollments.py:84]: filename has start and end dates

**Pull request merged**